### PR TITLE
More robust config options

### PR DESCRIPTION
**Pull request by november781**
Added config options for Corporate and Private maps, Name option, and properly configed redis (I think)

What is needed at top level:
rebuild docker hub image with the new entrypoint and fix the docker-compose.yaml to use new server image.

### CODE REVIEW

**Comment by november781** [comment]
Solves Issue #10

File: Dockerfile [hunk 1]
```diff
@@ -31,10 +31,11 @@ RUN apt-get install -y \
 # COPY PATHFINDER
 ARG VERSION 
 RUN mkdir /var/www/pathfinder
-RUN git clone --branch $VERSION https://github.com/exodus4d/pathfinder.git /var/www/pathfinder
+RUN git clone https://github.com/exodus4d/pathfinder.git /var/www/pathfinder
```
Comment (porowns) [inline Dockerfile]: Why did you take out the branch checkout?

File: Dockerfile [hunk 2]
```diff
@@ -5,7 +5,7 @@ ENV DEBIAN_FRONTEND noninteractive
 RUN adduser --disabled-password --gecos '' pathfinder 
 
 # INSTALL PACKAGES
-RUN apt-get update && \
+RUN apt-get update --fix-missing&& \
```
Comment (porowns) [inline Dockerfile:8]: Why did you take this out? Needed to support container on Mac/etc. Was this breaking something for you?

File: Dockerfile [hunk 3]
```diff
@@ -61,6 +62,8 @@ RUN crontab /home/default_crontab
 COPY entrypoint.sh /usr/local/bin/
 RUN chmod +x /usr/local/bin/entrypoint.sh 
 
+RUN echo "Hello there!"
```
Comment (porowns) [inline Dockerfile]: Leftover debug code

File: config/redis.conf [hunk 4]
```diff
@@ -450,7 +450,7 @@ slave-priority 100
 # limit for maxmemory so that there is some free RAM on the system for slave
 # output buffers (but this is not needed if the policy is 'noeviction').
 #
-# maxmemory <bytes>
+maxmemory 64mb
```
Comment (porowns) [inline config/redis.conf:453]: Is this the recommended amount via Pathfinder?

File: docker-compose.yml [hunk 5]
```diff
@@ -12,7 +12,8 @@ services:
           MYSQL_USER: "${MYSQL_USER}"
 
     pathfinder:
-        image: kryptedgaming/pathfinder:latest
+        image: pathfinder:latest
```
Comment (porowns) [inline docker-compose.yml]: Not accepting changes to `docker-compose.yml`, make sure you keep your local version uncommited. Ideally we want users to pull from the docker image at Docker Hub

**Review by porowns — CHANGES_REQUESTED** [review]

**Comment by porowns** [comment]
Overall pretty good job on this, just need a few changes and we can merge it in :D

File: Dockerfile (hunk 1 above)
Comment (november781) [reply Dockerfile]: It was causing failed clones when I tried to build the image on my end

File: Dockerfile (hunk 2 above)
Comment (november781) [reply Dockerfile:8]: that should not have been removed....

File: config/redis.conf (hunk 4 above)
Comment (november781) [reply config/redis.conf:453]: Yes, but it does not seem to take, at least PF does not see it, Redis might need a config file reload before starting

File: Dockerfile (hunk 2 above)
Comment (november781) [reply Dockerfile:8]: it was not really removed, just changed to add the --fix-missing so I could update packages properly

File: docker-compose.yml (hunk 5 above)
Comment (november781) [reply docker-compose.yml]: yep, sorry that was debug stuffs left over.

**Review by november781 — COMMENTED** [review]
Fixed some of the requested changes, noted where others were changed to unbreak something

**Review by november781 — COMMENTED** [review]
Fixed some of the requested changes, noted where others were changed to unbreak something

File: Dockerfile (hunk 1 above)
Comment (porowns) [reply Dockerfile]: You need to pass an argument in. `docker build <whatever> --build-arg VERSION=master` Alternatively you can use args in docker-compose like so:
```
  args:
    - buildno=1
    - gitcommithash=cdc3b1
```

Comment (porowns) [reply Dockerfile]: This definitely needs to stay so that we can provide images for previous versions (e.g 1.5.4)

File: config/redis.conf (hunk 4 above)
Comment (porowns) [reply config/redis.conf:453]: Might need to add it to the entrypoint script

**Review by november781 — COMMENTED** [review]
Branch version flag added back

**Review by porowns — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (november781) [reply Dockerfile]: that --branch $Version results in an error when trying to build a new container (until the official on is updated)
it gives me /var/www/pathfinder is not a repository which is odd

**Pull request merged**